frontend/src/hex/HexView.svelte

```diff
@@ -95,23 +100,11 @@
     windowSize = chunkSize * 10,
     windowPadding = 1024;
 
-  $: dataLenPromise.then((r) => {
-    dataLength = r;
-  });
-
   $: childRangesPromise.then((r) => {
     childRanges = r;
   });
 
-  $: chunkDataPromise.then((r) => {
-    chunks = r;
-  });
-
-  $: childRangesPromise = calculateRanges(
-    $selectedResource,
-    dataLenPromise,
-    $settings.colors
-  );
+  $: childRangesPromise = calculateRanges($selectedResource, $settings.colors);
```

Comment:
This should probably still take `$dataLength` as an input so that it is reactively called when the store's value updates.